### PR TITLE
docker-compose: add otel-collector by default, disable jaeger by default

### LINKED ISSUE #40455: deploy-sourcegraph-docker: make Jaeger optional, always deploy otel-collector

### PR DESCRIPTION
**Pull request by bobheadxi**
Removes Jaeger from the default docker-compose deployment, and replaces it with OpenTelemetry Collector configured to log data only.

With the `jaeger/docker-compose.yaml` overlay, a Jaeger instance can be deployed and `otel-collector` will be configured to send traces to the deployed Jaeger instance.

The pure-docker version of this change is in https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/849

Closes https://github.com/sourcegraph/sourcegraph/issues/40455

### Checklist

* [x] Sister [deploy-sourcegraph](https://github.com/sourcegraph/deploy-sourcegraph) change: https://github.com/sourcegraph/deploy-sourcegraph/pull/4163
* [x] All images have a valid tag and SHA256 sum

### Test plan

```sh
docker-compose \
    -f docker-compose/docker-compose.yaml \
    -f docker-compose/jaeger/docker-compose.yaml \
    -f docker-compose/dev/docker-compose.yaml up
```

```sh
{
    "observability.tracing": { "type": "opentelemetry" }
}
```

Screenshot 2022-08-25 at 2 40 23 PM

### CODE REVIEW

**Review by sanderginn — APPROVED** [review]
It all looks good to me.

Is this repository meant to serve as a public deployment method? If so, we might want to offer means to set a custom otel config.

**Comment by bobheadxi** [comment]
> Is this repository meant to serve as a public deployment method? If so, we might want to offer means to set a custom otel config.

Yes - I added a mounted-by-default template similar to your approach in https://github.com/sourcegraph/deploy-sourcegraph/pull/4163: https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/848/commits/0b2f37874fe14c8ca879c6e645131b6db5f6ec82

**Review by sanderginn — APPROVED** [review]
LGTM

**Review by burmudar — APPROVED** [review]

File: docker-compose/jaeger/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - '0.0.0.0:6831:6831'
+      - '0.0.0.0:6832:6832'
+    networks:
+      - sourcegraph
+    restart: always
+    command: ['--memory.max-traces=20000']
+    environment:
+      - 'SAMPLING_STRATEGIES_FILE=/etc/jaeger/sampling_strategies.json'
+
+  # Configure collector to send traces to Jaeger
```
Comment (burmudar) [inline docker-compose/jaeger/docker-compose.yaml:32]:
```suggestion
  # Configure collector to send traces to Jaeger
  # By default listens on the following ports:
  # - grpc 4317
  # - http 4318
```

Comment (bobheadxi) [reply docker-compose/jaeger/docker-compose.yaml:32]: This is already indicated on docstring for the underlying service configuration in the base `docker-compose.yaml` file, so I think we should be fine to omit this here

**Comment by michaellzc** [comment]
docker compose stuff looks good to me, but how about our pure docker distro? https://github.com/sourcegraph/deploy-sourcegraph-docker/tree/master/pure-docker

I know there has not been much clear signal how to better support pure-docker distro in the future, but I do think we should back-port these changes to pure-docker as well.

**Comment by bobheadxi** [comment]
pure-docker changes are here: https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/849 with some slight variations

**Review by michaellzc — APPROVED** [review]

**Pull request merged**